cloud-sql/mysql/sqlalchemy/main.py

```diff
@@ -68,39 +61,51 @@ def init_connection_engine():
 
 def init_tcp_connection_engine(db_config):
     # [START cloud_sql_mysql_sqlalchemy_create_tcp]
-    db_socket_addr = os.environ.get("DB_HOST").split(":")
+    # Remember - storing secrets in plaintext is potentially unsafe. Consider using
+    # something like https://cloud.google.com/kms/ to help keep secrets secret.
+    db_user = os.environ.get("DB_USER")
+    db_pass = os.environ.get("DB_PASS")
+    db_name = os.environ.get("DB_NAME")
+    db_host = os.environ.get("DB_HOST")
+
+    db_host_parts = db_host.split(":")
 
-    # Extract host and port from socket address
-    db_host = db_socket_addr[0]
-    db_port = int(db_socket_addr[1])
+    # Extract host and port from db_host socket address
+    db_hostname = db_host_parts[0]
+    db_port = int(db_host_parts[1])
 
-    return sqlalchemy.create_engine(
+    pool = sqlalchemy.create_engine(
         # Equivalent URL:
         # mysql+pymysql://<db_user>:<db_pass>@<db_host>:<db_port>/<db_name>
         sqlalchemy.engine.url.URL(
             drivername="mysql+pymysql",
             username=db_user,
             password=db_pass,
-            host=db_host,
+            host=db_hostname,
             port=db_port,
             database=db_name,
         ),
         # ... Specify additional properties here.
-        # [START_EXCLUDE]
+        # [END cloud_sql_mysql_sqlalchemy_create_tcp]
         **db_config
-        # [END_EXCLUDE]
+        # [START cloud_sql_mysql_sqlalchemy_create_tcp]
     )
     # [END cloud_sql_mysql_sqlalchemy_create_tcp]
 
+    return pool
+
 
 def init_unix_connection_engine(db_config):
     # [START cloud_sql_mysql_sqlalchemy_create_socket]
-    if os.environ.get("DB_SOCKET_PATH"):
-        socket_path = os.environ.get("DB_SOCKET_PATH")
-    else:
-        socket_path = "/cloudsql"
-
-    return sqlalchemy.create_engine(
+    # Remember - storing secrets in plaintext is potentially unsafe. Consider using
+    # something like https://cloud.google.com/kms/ to help keep secrets secret.
+    db_user = os.environ.get("DB_USER")
+    db_pass = os.environ.get("DB_PASS")
+    db_name = os.environ.get("DB_NAME")
```

Comment:
Let's make these required, since the application will fail without them: 
```suggestion
    db_user = os.environ["DB_USER"]
    db_pass = os.environ["DB_PASS"]
    db_name = os.environ["DB_NAME"]
```

Reply:
Thanks. I'm making all the env vars except the "DB_SOCKET_DIR" required. Including "DB_HOST" for TCP connections and "CLOUD_SQL_CONNECTION_NAME" for Unix connections.